lib/src/select/Select.tsx

```diff
@@ -432,15 +314,52 @@ const DxcSelect = React.forwardRef<RefType, SelectPropsType>(
                 </GroupLabel>
                 {option.options.map((singleOption) => {
                   globalIndex++;
-                  return <Option option={singleOption} index={globalIndex} isGroupedOption={true} />;
+                  return (
+                    <Option
+                      id={`option-${globalIndex}`}
+                      option={singleOption}
+                      onClick={() => {
+                        handleSelectChangeValue(singleOption);
+                        !multiple && closeOptions();
+                        setSearchValue("");
+                      }}
+                      multiple={multiple}
```

Comment:
Following the comment above, we can call `handleOptionOnClick(singleOption)` here.